### PR TITLE
Adding Typescript to the project and converting all JS to valid TS

### PR DESCRIPTION
**Pull request by xnf**
- [x] Install TS/TSLint with config
- [x] Convert JS to TS
- [x] Create scripts in package.json for building (and prebuild linting)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (trioletas) [inline LICENSE]: restore the license please,we need it on GitHub

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,100 @@
+{
```
Comment (trioletas) [inline tslint.json]: please use one of the presets, such as ` "tslint:recommended"`
override where necessary

Comment (trioletas) [reply tslint.json]: we can discuss particular rules that we care about

File: server/server.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+var loopback = require('loopback');
```
Comment (trioletas) [inline server/server.ts]: prefer es6 module imports over require

File: server/server.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+var boot = require('loopback-boot');
+
+var app = module.exports = loopback();
```
Comment (trioletas) [inline server/server.ts]: prefer es6 module exports over commonjs modules

File: server/server.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+app.start = function () {
```
Comment (trioletas) [inline server/server.ts]: either name the function or use an arrow function

File: server/server.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    if (app.get('loopback-component-explorer')) {
+      var explorerPath = app.get('loopback-component-explorer').mountPath;
+      console.log('Browse your REST API at %s%s', baseUrl, explorerPath);
+    }
+  });
+};
+
+// Bootstrap the application, configure models, datasources and middleware.
+// Sub-apps like REST API are mounted via boot scripts.
+boot(app, __dirname, function (err) {
```
Comment (trioletas) [inline server/server.ts]: can we use async await here ? avoid callbacks when possible

File: package.json [hunk 7]
```diff
@@ -3,11 +3,11 @@
...
-    "lint": "eslint .",
-    "start": "node .",
-    "posttest": "npm run lint && nsp check"
+    "start": "node ./build/server.ts",
+    "lint": "tslint -t stylish 'server/**/*.ts'"
   },
   "dependencies": {
+    "@types/loopback": "3.2.1",
     "loopback": "3.14.0",
     "loopback-boot": "2.26.2",
```
Comment (trioletas) [inline package.json]: pls also add `@types/loopback-boot`

**Comment by xnf** [comment]
Kind of done-ish. Missing features:

1. build JS in separate folder (need to complete more taks to see how to build it the best way in order not to overkill with webpack, for instance)
2. Loopback types do not seem to work correctly. For instance - the server.js example from their website has "app.emit()" although types do not have "emit" neither directly or inherited.

File: LICENSE (hunk 1 above)
Comment (xnf) [reply LICENSE]: I can remove "C.T.Co Developers Community" part and leave"only "Copyright (c) 2017". Should I?

File: tslint.json (hunk 2 above)
Comment (xnf) [reply tslint.json]: i put  "tslint:recommended"  + no-console:false as this is backend app and console is useful

Comment (trioletas) [reply tslint.json]: awesome, thank you! We can drop console one once the logging framework is introduced :)

**Comment by trioletas** [comment]
Please also include incremental compilation, see the graphql templates on implementation details

File: LICENSE (hunk 1 above)
Comment (trioletas) [reply LICENSE]: No, please leave the license as is

**Pull request merged**